worlds/yugioh06/__init__.py

```diff
@@ -112,6 +116,14 @@ class Yugioh06World(World):
     item_name_groups = {
         "Core Booster": core_booster,
         "Campaign Boss Beaten": ["Tier 1 Beaten", "Tier 2 Beaten", "Tier 3 Beaten", "Tier 4 Beaten", "Tier 5 Beaten"],
+        "Challenge": challenges,
+        "Tier 1 Opponent": tier_1_opponents,
+        "Tier 2 Opponent": tier_2_opponents,
+        "Tier 3 Opponent": tier_3_opponents,
+        "Tier 4 Opponent": tier_4_opponents,
+        "Tier 5 Opponent": tier_5_opponents,
+        "Campaign Opponent": tier_1_opponents + tier_2_opponents + tier_3_opponents +
+                             tier_4_opponents + tier_5_opponents
     }
```

Comment:
```suggestion
    item_name_groups: Dict[str, Set[str]] = {
        "Core Booster": set(core_booster),
        "Campaign Boss Beaten": {"Tier 1 Beaten", "Tier 2 Beaten", "Tier 3 Beaten", "Tier 4 Beaten", "Tier 5 Beaten"},
        "Challenge": set(challenges),
        "Tier 1 Opponent": set(tier_1_opponents),
        "Tier 2 Opponent": set(tier_2_opponents),
        "Tier 3 Opponent": set(tier_3_opponents),
        "Tier 4 Opponent": set(tier_4_opponents),
        "Tier 5 Opponent": set(tier_5_opponents),
        "Campaign Opponent": set(tier_1_opponents + tier_2_opponents + tier_3_opponents +
                             tier_4_opponents + tier_5_opponents)
    }
```
Just to make it more convenient (and because I feel dumb for missing that these were lists)